src/main/java/uk/nhs/hee/tis/trainee/sync/facade/PlacementEnricherFacade.java

```diff
@@ -179,6 +211,73 @@ private void syncPlacement(Placement placement, String employingBodyName,
     tcsSyncService.syncRecord(placement);
   }
 
+  /**
+   * Enrich the placement with the given site name and location and then sync it.
+   *
+   * @param placement         The placement to sync.
+   * @param siteName The site name to enrich with.
+   * @param siteLocation  The site location to enrich with.
+   */
+
+  private void syncPlacementSiteDetails(Placement placement, String siteName,
+                                        String siteLocation) {
+    // Add extra data to placement data.
+    Map<String, String> placementData = placement.getData();
+    if (Strings.isNotBlank(siteName)) {
+      placementData.put(PLACEMENT_DATA_SITE_NAME, siteName);
+    }
+    if (Strings.isNotBlank(siteLocation)) {
+      placementData.put(PLACEMENT_DATA_SITE_LOCATION, siteLocation);
+    }
+
+    // Set the required metadata so the record can be synced using common logic.
+    placement.setOperation(LOAD);
+    placement.setSchema("tcs");
+    placement.setTable("Placement");
+
+    tcsSyncService.syncRecord(placement);
+  }
```

Comment:
This needs building in to `syncPlacementDetails`, everything must be synced together after building the complete record.

Reply:
Thanks - yes, I was wondering about that! :)